cleanreg.py

```diff
@@ -547,11 +571,22 @@ def get_deletiontags(verbose, tags_dates_digests, repo, repo_count):
     else:
         ammount_tags = len(all_tags)
 
+    if repo_count is None:
+        repo_count = 0
+
     if verbose > 1:
         print "Repo {0}: ammount_tags : {1}; repo_count: {2}".format(repo, ammount_tags, repo_count)
 
     if ammount_tags > repo_count:
         deletion_tags = collections.OrderedDict(islice(all_tags.iteritems(), ammount_tags - repo_count))
+        if regex is not None and regex != "_" and regex != "":
+            deletion_tags = {k: deletion_tags[k] for k in deletion_tags if not re.match(regex, k)}
+        if date is not None and date != "_" and date != "":
+            parsed_date = datetime.strptime(date, '%d.%m.%Y')
+            for tag in deletion_tags.keys():
+                tag_date = datetime.strptime(deletion_tags[tag]['date'].split('T')[0], '%Y-%m-%d')
```

Comment:
Better interpret the full datetime string (with the T). You can search for ISO Datetime